packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/DeclareLocalFunctionStatement.cs

```diff
@@ -8,8 +8,23 @@
 
 namespace Microsoft.Generator.CSharp.Statements
 {
-    public sealed record DeclareLocalFunctionStatement(CodeWriterDeclaration Name, IReadOnlyList<ParameterProvider> Parameters, CSharpType ReturnType, ValueExpression? BodyExpression, MethodBodyStatement? BodyStatement) : MethodBodyStatement
+    public sealed class DeclareLocalFunctionStatement : MethodBodyStatement
```

Comment:
Why was this changed from record?

Reply:
well they do not have to be records because we kind of never need those record features, such as:
- auto generated `ToString` method
- auto generated `Equals` method
one thing we might be using
- auto generated `GetHashCode` method

so changing back to class does not really change anything.
For debugging purpose (because the auto generated ToString might help in debugging purpose), we should try to support it via adding `DebuggerDisplay` attribute on the base type of everything to invoke the write method.

Therefore I am fine with this change.

Reply:
Records are great if you truly just want an immutable object with default everything.  I find them very restrictive when you want to start customizing access on getters / setters so swapped to classes.

Reply:
A simple example of this is when you want the property to be non nullable and you want the param to the ctor to be nullable / optional.